Fix TypeError in NetCommandFactoryJson.make_get_thread_stack_message()

jgru opened the pull request:
Dear maintainers, 

I experienced the following error using `debugpy` with Python3 in Version 3.9.2 both on macOS 12.3.1 and Debian Bookworm. 
The issue came up when using Emacs' `dap-mode`. 
```
Traceback (most recent call last):
  File "/home/username/.local/lib/python3.9/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py", line 252, in _on_run
    self.process_net_command_json(self.py_db, json_contents)
  File "/home/username/.local/lib/python3.9/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py", line 193, in process_net_command_json
    cmd = on_request(py_db, request)
  File "/home/username/.local/lib/python3.9/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_process_net_command_json.py", line 953, in on_stacktrace_request
    self.api.request_stack(py_db, request.seq, thread_id, fmt=fmt, start_frame=start_frame, levels=levels)
  File "/home/username/.local/lib/python3.9/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_api.py", line 206, in request_stack
    if internal_get_thread_stack.can_be_executed_by(get_current_thread_id(threading.current_thread())):
  File "/home/username/.local/lib/python3.9/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py", line 654, in can_be_executed_by
    self._cmd = py_db.cmd_factory.make_get_thread_stack_message(
  File "/home/username/.local/lib/python3.9/site-packages/debugpy/_vendored/pydevd/_pydevd_bundle/pydevd_net_command_factory_json.py", line 284, in make_get_thread_stack_message
    end = min(start + levels, total_frames)
TypeError: unsupported operand type(s) for +: 'NoneType' and 'int'
```
The error occured with the current HEAD of main-branch (commit 0a40cab) and with the version 1.6.0  hosted on PyPi. 
The cause of it seems to be that `start_frame` is used for addition in order to determine a stackframe number, although it is was None. 

This PR fixes the issue. 

Thanks already in advance for considering this small contribution. 

Best regards, 
jgru

fabioz commented:
@jgru thanks for the pull request (sorry for the delay in getting back to you).

I think that a better approach could be not letting the `None` get there in the first place (it should be an `int` at that point, not `Optional[int]`).

So, the fix should probably be changing 

`_pydevd_bundle.pydevd_process_net_command_json.PyDevJsonCommandProcessor.on_stacktrace_request`

Where it should check `startFrame` and `levels` for `None` and change to `0` if needed.

jgru commented:
> @jgru thanks for the pull request (sorry for the delay in getting back to you).
> 

Thanks for getting back to me.
> So, the fix should probably be changing
> `_pydevd_bundle.pydevd_process_net_command_json.PyDevJsonCommandProcessor.on_stacktrace_request`
> Where it should check `startFrame` and `levels` for `None` and change to `0` if needed.

I reverted the initial change and introduced two predicates in `_pydevd_bundle.pydevd_process_net_command_json.PyDevJsonCommandProcessor.on_stacktrace_request` to ensure that they will never be `None`. Hope this change is fine now. 

Best regards,
jgru

fabioz commented:
Seems good to me. Just merged.

Thank you very much for the fix.
